protected/tests/unit/DatasetLogTest.php

```diff
@@ -3,33 +3,37 @@
 
 class DatasetLogTest extends CDbTestCase
 {
-    public function testDatasetLogEntryFactory()
+    protected $fixtures=array(
+        'log'=>'DatasetLog',
+    );
+
+    public function testDataSetLogEntryFactory()
     {
-        $datasetId = 8;
+        $datasetId = 1;
         $fileName = "Pygoscelis_adeliae.scaf.fa.gz";
         $fileModel = "FileAttributes";
         $modelId = 11020;
         $fileId = 664;
         $datasetlog = DatasetLog::datasetLogEntryFactory($datasetId, $fileName, $fileModel, $modelId, $fileId);
         $this->assertNotNull($datasetlog);
         $this->assertTrue(is_a($datasetlog, DatasetLog::class));
-        $this->assertEquals($datasetId, $datasetlog->dataset_id);
-        $this->assertEquals("$fileName: file attribute deleted", $datasetlog->message);
-        $this->assertEquals($fileModel, $datasetlog->model);
-        $this->assertEquals($modelId, $datasetlog->model_id);
-        $this->assertEquals("./bin/adminFile/update/id/$fileId", $datasetlog->url);
+        $this->assertEquals($datasetlog->dataset_id, $this->log(3)->dataset_id);
+        $this->assertEquals($datasetlog->message, $this->log(3)->message);
+        $this->assertEquals($datasetlog->model, $this->log(3)->model);
+        $this->assertEquals($datasetlog->model_id, $this->log(3)->model_id);
+        $this->assertEquals($datasetlog->url, $this->log(3)->url);
     }
 
-    public function testCreateDatasetLogEntry()
-    {
-        $datasetId = 8;
-        $fileName = "Pygoscelis_adeliae.scaf.fa.gz";
-        $fileModel = "FileAttributes";
-        $modelId = 11020;
-        $fileId = 664;
-        $datasetlog = new DatasetLog(); //Instantiate a new Object
-        $result = $datasetlog->createDatasetLogEntry($datasetId, $fileName, $fileModel, $modelId, $fileId);
-        $this->assertTrue(true===$result, "No new entry in dataset log table");
-        $this->assertTrue($datasetlog->isNewRecord);
-    }
+//    public function testCreateDatasetLogEntry()
+//    {
+//        $datasetId = 1;
+//        $fileName = "Pygoscelis_adeliae.scaf.fa.gz";
+//        $fileModel = "FileAttributes";
+//        $modelId = 11020;
+//        $fileId = 664;
+//        $datasetlog = new DatasetLog(); //Instantiate a new Object
+//        $result = $datasetlog->createDatasetLogEntry($datasetId, $fileName, $fileModel, $modelId, $fileId);
+//        $this->assertTrue(true===$result, "No new entry in dataset log table");
+//        $this->assertTrue($datasetlog->isNewRecord);
```

Comment:
@pli888 @rija This test thrown the following error 
```
CDbException: CDbCommand failed to execute the SQL statement: SQLSTATE[23505]: Unique violation: 7 ERROR:  duplicate key value violates unique constraint "dataset_log_pkey"
DETAIL:  Key (id)=(1) already exists.
```
The mockup data should be `id=4`, so no duplication should be the case. I will try to figure it out.

Reply:
@kencho51 `testDataSetLogEntryFactory()` is a bit confusing. What seems to be currently happening is that it is creating a new `DatasetLog` record using the `datasetLogEntryFactory()` function then you are comparing this new `DatasetLog` record's attributes with those from the 4th array in `dataset_log.php`:
```
array(
         'id'=>4,
         'dataset_id'=>1,
         'message'=>"Pygoscelis_adeliae.scaf.fa.gz: file attribute deleted",
         'created_at'=>"2020-10-27 16:36:23.106366",
         'model'=>"FileAttributes",
         'model_id'=>11020,
         'url'=>"./bin/adminFile/update/id/664",
     ),
```

I think in this case you don't need to have any fixtures for this test but need to check that the website application has saved the newly created `DatasetLog` record correctly.

I think the reason for `CDbException` error message is that after the DatasetLog fixtures have been created in `DatasetLog` table then the primary key sequence responsible for creating new ids in `DatasetLog` table becomes out of sync.